### PR TITLE
Use the Kubernetes API for namespaces check

### PR DESCRIPTION
**Pull request by rthallisey**
**Describe what this PR does and why we need it**:
Verify that a namespace exists using the kubernetes API.

Changes proposed in this pull request
 - Fixes issue in k8s where projects don't exist

### CODE REVIEW

**Comment by shawn-hurley** [comment]
NACK, it is assumed that the project is created I think this is a valid check IMO. 

This is probably a pretty good place to determine if that assumption makes sense.

**Comment by rthallisey** [comment]
You can't create a service instance without a valid namespace/project.  I don't think there's a scenario where this code will be false.  Let me know if you have one in mind.

```
[rhallisey@rhev-i16c-04 ansible-service-broker]$ oc create -f  scripts/broker-ci/mediawiki123.yaml 
Error from server (NotFound): error when creating "scripts/broker-ci/mediawiki123.yaml": namespaces "non-existent-namespace" not found
```

**Comment by shawn-hurley** [comment]
When someone hits the endpoint without going through the catalog. We are
just a spec conforming thing, we should not assume our callers are doing
the right thing IMO.

**Comment by rthallisey** [comment]
I think this check came from the asbcli era which used the scenario you mentioned.  There isn't any harm in keeping another check around, but I think the scenario where this check fails is going to be rare. I have the patch to convert to a namespace already around. I'll swap to that one.

File: pkg/apb/provision_or_update.go [hunk 1]
```diff
@@ -49,11 +52,17 @@ func provisionOrUpdate(
...
+
 	ns := instance.Context.Namespace
-	log.Info("Checking if project %s exists...", ns)
-	if !projectExists(ns) {
-		log.Errorf("Project %s does NOT exist! Cannot provision requested %s", ns, instance.Spec.FQName)
-		return "", nil, fmt.Errorf("Project %s does not exist", ns)
+	log.Info("Checking if namespace %s exists.", ns)
+	_, err = k8scli.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+	if err != nil {
+		return "", nil, errors.New(fmt.Sprintf("Project %s does NOT exist!", ns))
```
Comment (djzager) [inline pkg/apb/provision_or_update.go]:
```
=================================
              Lint               
=================================
pkg/apb/provision_or_update.go:65:19: should replace errors.New(fmt.Sprintf(...)) with fmt.Errorf(...)
Found 1 lint suggestions; failing.
Makefile:27: recipe for target 'lint' failed
make: *** [lint] Error 1
```

File: pkg/apb/provision_or_update.go [hunk 2]
```diff
@@ -49,11 +52,17 @@ func provisionOrUpdate(
...
+
 	ns := instance.Context.Namespace
-	log.Info("Checking if project %s exists...", ns)
-	if !projectExists(ns) {
-		log.Errorf("Project %s does NOT exist! Cannot provision requested %s", ns, instance.Spec.FQName)
-		return "", nil, fmt.Errorf("Project %s does not exist", ns)
+	log.Info("Checking if namespace %s exists.", ns)
+	_, err = k8scli.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+	if err != nil {
+		return "", nil, fmt.Errorf("Project %s does NOT exist!", ns)
```
Comment (djzager) [inline pkg/apb/provision_or_update.go]: One more :wink: 
```
pkg/apb/provision_or_update.go:65:30: error strings should not be capitalized or end with punctuation or a newline
```

You can use `make check` to run through most of the tests before pushing.

Comment (rthallisey) [reply pkg/apb/provision_or_update.go]: I'll move back to what was there :).

**Comment by rthallisey** [comment]
thanks @djzager

**Pull request merged**